internal/pkg/docker/dockercompose/svc_image.go

```diff
@@ -0,0 +1,140 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package dockercompose
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/compose-spec/compose-go/types"
+)
+
+// convertImageConfig takes in a Compose build config & image field and returns a Copilot image manifest.
+func convertImageConfig(build *types.BuildConfig, labels map[string]string, imageLoc string) (manifest.Image, IgnoredKeys, error) {
+	image := manifest.Image{
+		DockerLabels: labels,
+	}
+
+	// note: Compose allows both build & image to be specified, but image takes precedence.
+	// In Copilot, however, build and image are mutually exclusive.
+	if imageLoc != "" {
+		image.Location = aws.String(imageLoc)
+		return image, nil, nil
+	} else if build == nil {
+		return manifest.Image{}, nil, errors.New("missing one of `build` or `image`")
+	}
+
+	ignored, err := unsupportedBuildKeys(build)
+	if err != nil {
+		return manifest.Image{}, ignored, err
+	}
+
+	args, err := convertMappingWithEquals(build.Args)
+	if err != nil {
+		return manifest.Image{}, nil, fmt.Errorf("convert build args: %w", err)
+	}
+
+	buildArgs := manifest.DockerBuildArgs{
+		Context:    nilIfEmpty(build.Context),
+		Dockerfile: nilIfEmpty(build.Dockerfile),
+		Args:       args,
+		Target:     nilIfEmpty(build.Target),
+		CacheFrom:  build.CacheFrom,
+	}
+
+	image.Build = manifest.BuildArgsOrString{
+		BuildArgs: buildArgs,
+	}
+
+	return image, ignored, nil
+}
+
+func nilIfEmpty(s string) *string {
+	if s == "" {
+		return nil
+	} else {
+		return &s
+	}
+}
+
+// convertMappingWithEquals checks for entries with missing values and generates an error if any are found.
+func convertMappingWithEquals(inArgs types.MappingWithEquals) (map[string]string, error) {
+	args := map[string]string{}
+	var badArgs []string
+
+	for k, v := range inArgs {
+		if v == nil {
+			badArgs = append(badArgs, k)
+		} else {
+			args[k] = *v
+		}
+	}
+
+	if badArgs != nil {
+		return nil, fmt.Errorf("some entries are missing values and require user input, "+
+			"this is not supported in Copilot: %v", badArgs)
```

Comment:
maybe?: 
```suggestion
		return nil, fmt.Errorf("%s '%v' %s missing %s and %s user input", english.PluralWord(badArgs, "entry", "entries"), badArgs, english.PluralWord(badArgs, "is", "are"), english.PluralWord(badArgs, "a value", "values"), english.PluralWord(badArgs, "requires", "require"))
```

Reply:
Looks like a nice change! I'll keep the "not supported" part though I think.